evalcache.plugin.zsh

```diff
@@ -7,28 +7,38 @@
 export ZSH_EVALCACHE_DIR=${ZSH_EVALCACHE_DIR:-"$HOME/.zsh-evalcache"}
 
 function _evalcache () {
-  local cmdHash="nohash"
+  local cmdHash="nohash" data="$*" name
+
+  for name in $@; do
```

Comment:
borrowing the logic from @fmbarina's PR #8 to ensure inline environment variable definitions don't end up being used as the name for the output file

Reply:
also, @fmbarina - if you like, I can remove this and keep the `name` logic on your PR, since otherwise I've obviated your PR (I'm sorry - I got over-enthusiatic and stole your idea :)